docs/components/DataList/DataList.stories.mdx

```diff
@@ -23,3 +25,201 @@ item.
 If you need to add more information, consider using the
 [DataTable](?path=/story/components-lists-and-tables-datatable-docs--page)
 component instead.
+
+## Implementation guidelines
+
+Since DataList is a large component, we use a mixture of compound components and
+props to separate each part of the component.
+
+## Main wrapper
+
+DataList is the main component that holds everything and important props.
+
+<ArgsTable of={DataList} />
+
+## Layout
+
+DataList.Layout enables you to implement the layout of the list items. Keep in
+mind that DataList doesn't have a default layout so you have to have at least 1
+implemented.
+
+It also supports different `size` breakpoints so you can completely change the
+component it's using depending on the window size.
+
+It is also important to add the type of `DataListItem<typeof data>` on the
+`item` param in the layout so you can easily access the known data.
+
+```tsx
+// You can have 1 responsive layout
+<DataList>
+  <DataList.Layout>
+    {/* Declare the type of the item */}
+    {(item: DataListItem<MyDataType>) => (
+      <Grid>
+        <Grid.Cell size={{ xs: 12 md: 6 }}>{item.name}</Grid.Cell>
+        <Grid.Cell size={{ xs: 12 md: 6 }}>{item.email}</Grid.Cell>
+      </Grid>
+    )}
+  </DataList.Layout>
+</DataList>
+
+// Or you can have multiple layouts per breakpoint
+<DataList>
+  <DataList.Layout size="xs">
+    {(item: DataListItem<MyDataType>) => (
+      <Grid>
+        <Grid.Cell size={{ md: 6 }}>{item.name}</Grid.Cell>
+        <Grid.Cell size={{ md: 6 }}>{item.email}</Grid.Cell>
+      </Grid>
+    )}
+  </DataList.Layout>
+
+  <DataList.Layout size="md">
+    {(item: DataListItem<MyDataType>) => (
+      <Content>
+        <div>{item.name}</div>
+        <div>{item.email}</div>
+      </Content>
+    )}
+  </DataList.Layout>
+</DataList>
+```
+
+<ArgsTable of={DataList.Layout} />
+
+## Per item action(s)
```

Comment:
```suggestion
### Item actions
```

Maybe debatable, but I think just saying "item actions" maps better to the actual component